unittests/feed/ProposedTransactionFeedTests.cpp

```diff
@@ -0,0 +1,325 @@
+    ctx.restart();
+    ctx.run();
+    EXPECT_TRUE(receivedFeedMessage().empty());
+}
+
+TEST_F(FeedProposedTransactionTest, Count)
+{
+    testFeedPtr->sub(sessionPtr);
+    // repeat
+    testFeedPtr->sub(sessionPtr);
+    EXPECT_EQ(testFeedPtr->transactionSubcount(), 1);
+
+    auto const account1 = GetAccountIDWithString(ACCOUNT1);
+    testFeedPtr->sub(account1, sessionPtr);
+    // repeat
+    testFeedPtr->sub(account1, sessionPtr);
+    EXPECT_EQ(testFeedPtr->accountSubCount(), 1);
+
+    auto const sessionPtr2 = std::make_shared<MockSession>(tagDecoratorFactory);
+    testFeedPtr->sub(sessionPtr2);
+    EXPECT_EQ(testFeedPtr->transactionSubcount(), 2);
+
+    auto const account2 = GetAccountIDWithString(ACCOUNT2);
+    testFeedPtr->sub(account2, sessionPtr2);
+    EXPECT_EQ(testFeedPtr->accountSubCount(), 2);
+    testFeedPtr->sub(account1, sessionPtr2);
+    EXPECT_EQ(testFeedPtr->accountSubCount(), 3);
+
+    testFeedPtr->unsub(sessionPtr);
+    EXPECT_EQ(testFeedPtr->transactionSubcount(), 1);
+
+    // unsub unsubscribed account
+    testFeedPtr->unsub(account2, sessionPtr);
+    EXPECT_EQ(testFeedPtr->accountSubCount(), 3);
+
+    testFeedPtr->unsub(account1, sessionPtr);
+    EXPECT_EQ(testFeedPtr->accountSubCount(), 2);
+    testFeedPtr->unsub(account1, sessionPtr2);
+    EXPECT_EQ(testFeedPtr->accountSubCount(), 1);
+    testFeedPtr->unsub(account2, sessionPtr2);
+    EXPECT_EQ(testFeedPtr->accountSubCount(), 0);
+}
+
+TEST_F(FeedProposedTransactionTest, AutoDisconnect)
+{
+    testFeedPtr->sub(sessionPtr);
+    // repeat
+    testFeedPtr->sub(sessionPtr);
+    EXPECT_EQ(testFeedPtr->transactionSubcount(), 1);
+
+    auto const account1 = GetAccountIDWithString(ACCOUNT1);
+    testFeedPtr->sub(account1, sessionPtr);
+    // repeat
+    testFeedPtr->sub(account1, sessionPtr);
+    EXPECT_EQ(testFeedPtr->accountSubCount(), 1);
+
+    auto sessionPtr2 = std::make_shared<MockSession>(tagDecoratorFactory);
+    testFeedPtr->sub(sessionPtr2);
+    EXPECT_EQ(testFeedPtr->transactionSubcount(), 2);
+
+    auto const account2 = GetAccountIDWithString(ACCOUNT2);
+    testFeedPtr->sub(account2, sessionPtr2);
+    EXPECT_EQ(testFeedPtr->accountSubCount(), 2);
+    testFeedPtr->sub(account1, sessionPtr2);
+    EXPECT_EQ(testFeedPtr->accountSubCount(), 3);
+
+    sessionPtr2.reset();
+    EXPECT_EQ(testFeedPtr->accountSubCount(), 1);
+    EXPECT_EQ(testFeedPtr->transactionSubcount(), 1);
+
+    sessionPtr.reset();
+    EXPECT_EQ(testFeedPtr->accountSubCount(), 0);
+    EXPECT_EQ(testFeedPtr->transactionSubcount(), 0);
+}
+
+struct ProposedTransactionFeedMockPrometheusTest : WithMockPrometheus, SyncAsioContextTest {
+protected:
+    util::TagDecoratorFactory tagDecoratorFactory{util::Config{}};
+    std::shared_ptr<web::ConnectionBase> sessionPtr;
+    std::shared_ptr<ProposedTransactionFeed> testFeedPtr;
+
+    void
+    SetUp() override
+    {
+        WithMockPrometheus::SetUp();
```

Comment:
You don't need to do this. `WithMockPrometheus` does everything constructor/destructor. I hope in the future we will refactor tests to not have this SetUp()/TearDown() pattern everywhere.